run-cleanup.sh

```diff
@@ -21,30 +21,55 @@ if [[ -e "${CLEANED_REPOSITORY}" ]]; then
   exit 1
 fi
 
+ORDERED_REPOSITORY="$(mktemp -d)"
+echo "Storing intermediate reordered repository under ${ORDERED_REPOSITORY}"
+
 
 BASE="$(dirname "$(readlink -f "$0")")"
-SETUP_CLEANUP="${BASE}/setup-cleanup.sh"
+SETUP_MERCURIAL="${BASE}/setup-mercurial.sh"
 VIRTUALENV="${BASE}/data/py3-env"
 
-if ! /bin/bash "${SETUP_CLEANUP}"; then
+if ! /bin/bash "${SETUP_MERCURIAL}"; then
   echo "Error during setup."
   exit 2
 fi
 source "${VIRTUALENV}/bin/activate"
-
 # Disable all extensions.
 # (https://stackoverflow.com/questions/46612210/mercurial-disable-all-the-extensions-from-the-command-line)
-HGRCPATH= hg \
+export HGRCPATH=
+export HGPLAIN=
+
+echo "Looking for missing commits"
+
+if hg -R "${SRC_REPOSITORY}" incoming http://hg.fast-downward.org; then
```

Comment:
I just got the following output when I accidentally ran cleanup on a repository that was already cleaned:

```
Looking for missing commits
comparing with http://hg.fast-downward.org/
searching for changes
abort: repository is unrelated
Cloning official repository
```
It looks like the error (`abort: repository is unrelated`) still had the exit code of 0 (no commits to pull) and the script continued with cloning the official repository. Should we check for this?

Reply:
I wouldn't bother. This is not a script meant to be run by many people many times, and we cannot predict all user errors. I think users that make this error are likely to find out eventually what went wrong, so all this would do is potentially save them a bit of time.

Also, we should merge this soon; as far as I understand, the main branch is still on code that does no reordering and is not likely to work.